internal/tools/semconvkit/main.go

```diff
@@ -0,0 +1,83 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package semconvkit is used to generate opentelemetry-go specific semantic
+// convention code. It is expected to be used in with the semconvgen utility
+// (go.opentelemetry.io/build-tools/semconvgen) to completely generate
+// versioned sub-packages of go.opentelemetry.io/otel/semconv.
+package main
+
+import (
+	"embed"
+	"flag"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"text/template"
+)
+
+var (
+	out = flag.String("output", "./", "output directory")
+	tag = flag.String("tag", "", "OpenTelemetry tagged version")
+
+	//go:embed templates/*.tmpl
+	rootFS embed.FS
+)
+
+// SemanticConventions are information about the semantic conventions being
+// generated.
+type SemanticConventions struct {
+	// SemVer is the semantic version (i.e. 1.7.0 and not v1.7.0).
+	SemVer string
+	// TagVer is the tagged version (i.e. v1.7.0 and not 1.7.0).
+	TagVer string
```

Comment:
Nit. But since both values should always derive from the same, how about storing only one, and have the other be a method?

Reply:
TIL. Will do :+1: